libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -114,6 +121,10 @@ void AP_Scheduler::init(const AP_Scheduler::Task *tasks, uint8_t num_tasks, uint
     perf_info.set_loop_rate(get_loop_rate_hz());
     perf_info.reset();
 
+    if (_options & uint8_t(Options::RECORD_TASK_INFO)) {
```

Comment:
Do we still need to allocate here when you're allocating elsewhere too?

Reply:
I prefer doing it here as well, the update cycle only runs every 10s